networking/p2p/src/test/java/tech/pegasys/teku/networking/p2p/connection/ConnectionManagerTest.java

```diff
@@ -204,42 +203,51 @@ public void shouldConnectToKnownPeersWhenStarted() {
     when(network.connect(any(PeerAddress.class))).thenReturn(new SafeFuture<>());
 
     manager.start().join();
+    advanceTimeABit();
 
     verify(network).connect(PEER1);
     verify(network).connect(PEER2);
   }
 
   @Test
+  @Disabled("Shouldn't work, disabled until clarifying")
   public void shouldNotRetryConnectionsToDiscoveredPeersOnFailure() {
```

Comment:
I'm not entirely sure what needs clarifying here.  We shouldn't try to make a connection to a peer we've already connected to and seen fail - otherwise we just waste our time connecting to peers we already know won't work.

Reply:
I mean clarifying, how and when it was working. I guess it was sometihing using pools but not getting whole idea. It's not working in the current master

Reply:
I'm not sure what you mean by it's not working in the current master.  The test is enabled and passing in the current master. The `ConnectionManager.isPeerValid` method does the filtering to ensure we don't retry connections we shouldn't.

Reply:
- test cuts of on the future return in the master, it doesn't reach the point where we are attempting to connect to the discovered peers for the second time
- `ConnectionManager.isPeerValid` tests only fork currently

Reply:
Ah right, so we need to update the test to make sure it triggers the case.  This bit of filtering actually turns out to be done in `Eth2PeerSelectionStrategy.checkCandidate` so maybe the test we need should be for `Eth2PeerSelectionStrategy` rather than here.

Reply:
I see both cases are covered in `ReputationManagerTest`, just removed them from `ConnectionManagerTest`